lib/block-supports/layout.php

```diff
@@ -486,6 +486,10 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 		}
 	}
 
+	// Add combined layout and block classname for global styles to hook onto.
+	$block_name    = explode( '/', $block['blockName'] );
+	$class_names[] = 'wp-block-' . end( $block_name ) . '-' . $layout_classname;
```

Comment:
Good question! That could result in a mismatch between classnames in markup and CSS output, because the CSS rules are using [a similar logic](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/global-styles/use-global-styles-output.js#L911) to generate the block classname. 

Might the existence of classname filters have any impact on this functionality? I can't find any use of this filter in Core, do you know of any examples?

Reply:
> do you know of any examples?

No, I don't know of any examples where folks are using the filter. It could be fairly uncommon?

> That could result in a mismatch between classnames in markup and CSS output, because the CSS rules are using [a similar logic](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/global-styles/use-global-styles-output.js#L911) to generate the block classname.

If we switched to using `wp_get_block_default_classname`, I imagine we'd also switch to using `getBlockDefaultClassName( name )` in the JS logic, so there (at least theoretically) shouldn't be a mismatch. Both functions share the same logic, and it's pretty similar to the logic in this PR, I think, from a quick skim of the function definitions.

The main benefit of using those functions probably isn't really the filtering, though, it's probably just a small (and non-blocking) opportunity to re-use existing functions if it feels like it'd neaten things. But you do raise a good point about ensuring consistency between the JS and PHP output! I don't feel too strongly either way, as it'd be a quick change if we want to pick one approach and later flip to the other if we change our minds 🙂

Reply:
It would be fine to use in the layout JS logic, it's in the global styles output that things get dodgy because where we're generating the CSS rule around [here](https://github.com/WordPress/gutenberg/blob/add/cover-spacing/packages/block-editor/src/components/global-styles/use-global-styles-output.js#L398), the selector might not be a block at all. And if it is a block, its selector was generated using the [same logic](https://github.com/WordPress/gutenberg/blob/add/cover-spacing/packages/block-editor/src/components/global-styles/use-global-styles-output.js#L911) I used here 😅 

I'm not sure if that bit can be replaced by `wp_get_block_default_classname` without any side effects, might have to create a classname filter to test it.

Reply:
Ah gotcha! In that case, maybe it's better to stick with the rules you've got in this PR for now, if it looks like it's going to be either brittle or tricky to get it to work in the global styles logic 👍

Reply:
Just resolving this old thread — using the hard-coded logic in this PR sounds fine to me, and ensures there's consistency without filters messing around with things unexpectedly 😄